pennylane/ops/op_math/controlled_ops.py

```diff
@@ -176,12 +181,13 @@ def __init__(
         work_wires = Wires(() if work_wires is None else work_wires)
         control_wires = wires[:-1]  # default
 
-        if hasattr(base, "wires") and len(wires) != 0:
+        if isinstance(base, qml.QubitUnitary):
             warnings.warn(
-                "base operator already has wires; values specified through wires kwarg will be ignored."
+                "QubitUnitary input to ControlledQubitUnitary is deprecated and will be removed in v0.42. "
+                "Instead, please use a full matrix as input.",
+                qml.PennyLaneDeprecationWarning,
             )
-            # base should have the target_wires. Then control_wires should be the ones in wires that are not in base.wires
-            control_wires = [w for w in wires if w not in base.wires]
+            base = base.matrix()
 
         if isinstance(base, Iterable):
```

Comment:
What happens if `base` isn't an iterable?

Reply:
Oh yeah let me add error along with test in next round

Reply:
Fixed at https://github.com/PennyLaneAI/pennylane/pull/6840/commits/eb716895ebbe9335fdb1c2ae6be57c060b35b174